test/market/test_multichain.py

```diff
@@ -246,8 +247,40 @@ def test_update_block_with_beneficiary(self):
         self.assertEqualBlocks(result_benefactor, result)
         self.assertEqualBlocks(result_beneficiary, result)
 
+    def test_check_add_genesis_block(self):
+        """
+        This test checks the functionality of adding a block to an empty blockchain.
+        """
+
+        # Add genesis block to the blockchain
+        self.db.check_add_genesis_block(self.member_bank, '')
+        # Get the genesis block by the public key and sequence number
+        genesis_block = self.db.get_by_public_key_and_sequence_number(str(self.member_bank), 0)
```

Comment:
If you add the ability to create a genesis block separately you can check the "check_add_genesis_block" step, and just create a genesis block. Which you can use below to check if the new block that has been added automatically triggered the genesis block creation by checking if the previous block hash is equal to the genesis hash.